File: packages/widget_screenshot/ios/Classes/WidgetShotPlugin.swift
import Flutter
import UIKit

public class WidgetShotPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        // https://github.com/flutter/flutter/issues/118832
//        let taskQueue = registrar.messenger.makeBackgroundTaskQueue()
//        let channel = FlutterMethodChannel(name: "widget_shot",
//                                           binaryMessenger: registrar.messenger(),
//                                           codec: FlutterStandardMethodCodec.sharedInstance(),
//                                           taskQueue: taskQueue)
        let channel = FlutterMethodChannel(name: "widget_shot", binaryMessenger: registrar.messenger())
        let instance = WidgetShotPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        if call.method == "merge" {
            let arguments = call.arguments
            if arguments != nil {
                result(Merger(param: arguments! as! [String: Any]).merge())
            } else {
                result(nil)
            }
        }
        result(FlutterMethodNotImplemented)
    }
}

File: packages/widget_screenshot/android/src/main/kotlin/com/bingo/widgetshot/WidgetShotPlugin.kt
package com.bingo.widgetshot

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import io.flutter.plugin.common.StandardMethodCodec


/** ImageMergerPlugin */
class WidgetShotPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        val taskQueue =
            flutterPluginBinding.binaryMessenger.makeBackgroundTaskQueue()
        channel = MethodChannel(
            flutterPluginBinding.binaryMessenger,
            "widget_shot",
            StandardMethodCodec.INSTANCE,
            taskQueue
        )
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
        if (call.method == "merge") {
            val arguments = call.arguments as Map<String, Any>
            val merger = Merger(arguments)
            result.success(merger.merge())
        } else {
            result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
